restapi.go

```diff
@@ -2328,3 +2351,107 @@ func (s *Session) RelationshipsMutualGet(userID string) (mf []*User, err error)
 	err = unmarshal(body, &mf)
 	return
 }
+
+// ------------------------------------------------------------------------------------------------
+// Functions specific to application (slash) commands
+// ------------------------------------------------------------------------------------------------
+
+// ApplicationCommandCreate creates an global application command. If guild is specified - creates guild specific application command.
+func (s *Session) ApplicationCommandCreate(cmd *ApplicationCommand, guildID string) (rcmd *ApplicationCommand, err error) {
+	endpoint := EndpointApplicationGlobalCommands(s.State.User.ID)
```

Comment:
I'd prefer not to use `s.State.User.ID` here (or in the other similar functions). Passing in the application ID as a parameter is slightly more cumbersome, but it allows the functions to be used without an active gateway connection—a fairly common use-case now that interactions can be sent via HTTP webhooks (the library should still be useful to those users).

Reply:
Well, okay.

Reply:
But... can we somehow avoid additional passing of ID when It's not necessary. :thinking: